src/main/java/io/github/thebusybiscuit/slimefun4/core/multiblocks/MultiBlockMachine.java

```diff
@@ -165,6 +168,31 @@ public void load() {
         }
     }
 
+    /**
+     * This method handles an output {@link ItemStack} from the {@link MultiBlockMachine} which has a crafting delay
+     *
+     * @param outputItem
+     *          A crafted {@link ItemStack} from {@link MultiBlockMachine}
+     * @param block
+     *          Main {@link Block} of our {@link Container} from {@link MultiBlockMachine}
+     * @param blockInv
+     *          The {@link Inventory} of our {@link Container}
+     *
+     */
+    @ParametersAreNonnullByDefault
+    protected void handleCraftedItem(ItemStack outputItem, Block block, Inventory blockInv) {
+        Inventory outputInv = findOutputInventory(outputItem, block, blockInv);
+
+        if (outputInv != null) {
+            outputInv.addItem(outputItem);
+        } else if (InvUtils.fits(blockInv, outputItem)) {
+            blockInv.addItem(outputItem);
+        } else {
+            // fallback
+            SlimefunUtils.spawnItem(block.getLocation(), outputItem, ItemSpawnReason.MULTIBLOCK_MACHINE_OVERFLOW, true);
+        }
```

Comment:
Tiny improvement to improve performance even further, other than this I consider it approved.

```suggestion
        } else {
            ItemStack rest = blockInv.addItem(outputItem).get(0);

            // fallback: drop item
            if (rest != null) {
                SlimefunUtils.spawnItem(block.getLocation(), rest, ItemSpawnReason.MULTIBLOCK_MACHINE_OVERFLOW, true);
            }
        }
```